src/components/common/ratio/Ratio.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { useState } from "react";
+import "twin.macro";
+import "styled-components/macro";
+import Quantity from "../quantity/Quantity";
+import { CQL } from "cqm-models";
+
+interface RatioProps {
+  label: string;
+  numerator: CQL.Quantity;
+  denominator: CQL.Quantity;
+  canEdit: boolean;
+}
+
+const Ratio = ({ label, numerator, denominator, canEdit }: RatioProps) => {
+  const [currentNumerator, setCurrentNumerator] =
+    useState<CQL.Quantity>(numerator);
+  const [currentDenominator, setCurrentDenominator] =
```

Comment:
not sure we should need this internal state...shouldn't the value just come from the parent, and then when a value changes trigger the `onChange` with the modified `Ratio` object?

Reply:
removed internal state variables.